tests/sentry/test_event_manager.py

```diff
@@ -774,6 +775,51 @@ def test_environment(self):
 
         assert dict(event.tags).get('environment') == 'beta'
 
+    @mock.patch('sentry.event_manager.post_process_group.delay')
+    def test_group_environment(self, mock_post_process_group_delay):
+        def save_event():
+            manager = EventManager(self.make_event(**{
+                'event_id': uuid.uuid1().hex,  # don't deduplicate
+                'environment': 'beta',
+            }))
+            manager.normalize()
+            return manager.save(self.project.id)
+
+        event = save_event()
+
+        # Ensure the `GroupEnvironment` record was created.
+        GroupEnvironment.objects.get(
+            group_id=event.group_id,
+            environment_id=Environment.objects.get(
+                organization_id=self.project.organization_id,
+                name=event.get_tag('environment'),
+            ).id,
+        )
+
+        # Ensure that the first event in the (group, environment) pair is
+        # marked as being part of a new environment.
+        mock_post_process_group_delay.assert_called_with(
+            group=event.group,
+            event=event,
+            is_new=True,
+            is_sample=False,
+            is_regression=False,
+            is_new_group_environment=True,
+        )
+
+        event = save_event()
+
+        # Ensure that the next event in the (group, environment) pair is *not*
+        # marked as being part of a new environment.
+        mock_post_process_group_delay.assert_called_with(
+            group=event.group,
+            event=event,
+            is_new=False,
+            is_sample=False,
+            is_regression=None,  # XXX: wut
```

Comment:
I don't know why this is being passed as `None` instead of `False` (didn't investigate.)